src/rust/src/asn1.rs

```diff
@@ -71,22 +71,38 @@ pub(crate) fn py_uint_to_big_endian_bytes<'p>(
     py: pyo3::Python<'p>,
     v: pyo3::Bound<'p, pyo3::types::PyInt>,
 ) -> pyo3::PyResult<PyBackedBytes> {
-    if v.lt(0)? {
-        return Err(pyo3::exceptions::PyValueError::new_err(
-            "Negative integers are not supported",
-        ));
-    }
-
+    reject_negative_integer(&v)?;
```

Comment:
I don't think you need this call, since `py_uint_to_be_bytes_with_length` does the check